### ISSUE TITLE
File-preview should allow to customize crop ratios

### ISSUE THREAD

**Issue by FRSgit** [directus/app#1468]
# Feature Request

## What problem does this feature solve?
Admin panel user would want to crop image to exact ratio of some parts of his webpage. So it would be nice to have a possibility to setup custom ratios with some connected labels, e.g.:

```json
[
   { "label": "Img left", "value": "1.875" }
]
```

## How do you think this should be implemented?
Well, current `cropRatioOptions` object in `src\interfaces\file-preview\input.vue` should be rewritten as array including object like one written in example above and be merged with `options.ratios` (which will contain json array of custom ratios). Then, `setAspectRatio` method in the same file should accept ratio object as a parameter and stop doing any calculations for default switch case (all values will be already calculated in cropRatioOptions).

## Would you be willing to work on this?
Yup.

**Comment by benhaynes** [comment]
Love it!!

**Comment by FRSgit** [comment]
Of course it should be done after #1229, but hopefully that would be ready soon ;)

**Comment by rijkvanzanten** [comment]
To achieve better clarity/visibility, we are now tracking feature requests within the [Feature Request](https://github.com/directus/app/projects/45) project board.

**This issue being closed does not mean it's not being considered.**
